examples/server/server.cpp

```diff
@@ -2185,13 +2201,38 @@ struct server_context {
         // check if there is incomplete UTF-8 character at the end
         bool incomplete = validate_utf8(slot.generated_text) < slot.generated_text.size();
 
-        // search stop word and delete it
+
         if (!incomplete) {
             size_t pos = std::min(slot.n_sent_text, slot.generated_text.size());
 
             const std::string str_test = slot.generated_text.substr(pos);
             bool send_text = true;
 
+            // Handle the start strings
+            if (!slot.start_string_found && slot.has_next_token && !slot.params.start_strings.empty()) {
+                size_t max_start_string_size = slot.params.start_string_max_len;
+                size_t search_len = max_start_string_size + token_str.size();
+                size_t search_pos = 0;
+                if (slot.generated_text.size() > search_len) {
+                    search_pos = slot.generated_text.size() - search_len;
+                }
+
+                std::pair<size_t, int> search_result = find_first_substring(slot.generated_text,slot.params.start_strings, search_pos);
+                bool found = search_result.first != std::string::npos;
+
+                if (found) {
+                    auto found_pos = search_result.first;
+                    std::string found_string = slot.params.start_strings[search_result.second];
+                    slot.generated_text.erase(
+                        slot.generated_text.begin(),
+                        slot.generated_text.begin() + found_pos + found_string.size());
```

Comment:
If I understand correctly, the final thing we need is `found_pos + found_string.size()`, so probably `find_first_substring` don't even need to return the `found_pos` and `found_string`. Instead, it could return the position right **after** the found start string.

For example, my start string is `</think>`:

```
Ok I'm ready to give the answer</think>The final answer is
                                       ^ return this position
```